Source/Scene/Model.js

```diff
@@ -1209,9 +1211,14 @@ define([
         var basePath = defaultValue(options.basePath, modelResource.clone());
         var resource = Resource.createIfNeeded(basePath);
 
-        // If no cache key is provided, use the absolute URL, since two URLs with
-        // different relative paths could point to the same model.
-        var cacheKey = defaultValue(options.cacheKey, getAbsoluteUri(modelResource.url));
+        // If no cache key is provided, use a GUID.
+        // Check using a URI to GUID dictionary that we have not already added this model.
+        var cacheKey = defaultValue(options.cacheKey, defaultValue(uriToGuid[getAbsoluteUri(modelResource.url)], createGuid()));
+        var guidFromUri = uriToGuid[getAbsoluteUri(modelResource.url)];
+        if(!defined(guidFromUri)){
```

Comment:
We only want to do this if `options.cackeKey` is `undefined`, right?
I would do this instead:

```js
var cacheKey = defaultValue(options.cacheKey, uriToGuid[getAbsoluteUri(modelResource.url)]);
if (!defined(cacheKey)) {
    cacheKey = createGuid();
    uriToGuid[getAbsoluteUri(modelResource.url)] = cacheKey;
}
```

That way we only have to get the value of  `uriToGuid[getAbsoluteUri(modelResource.url)]` once and we only `createGuid` if we need to.